### PR TITLE
Allow modify AuthType when call login method 

### PR DESCRIPTION
**Pull request by sangpan**
Allow modify AuthType when call login method like this iOS version fix : 
https://github.com/mavris/FacebookFix

Can access to set auth Type by LoginButton or LoginManger.

        LoginButton mLoginButton = findViewById(R.id.login_button);
        mLoginButton.setAuthType("reauthorize");
or

            LoginManager manager = LoginManager.getInstance();
            manager.setAuthType("reauthorize");

Source from Master branch.

Thanks for proposing a pull request.

To help us review the request, please complete the following:
- [x] sign contributor license agreement: https://developers.facebook.com/opensource/cla
- [x] submit against our `:dev` branch, not `master`.
- [x] describe the change (for example, what happens before the change, and after the change)

### CODE REVIEW

**Comment by nanomech** [comment]
Nice. we need it!

File: facebook-common/src/main/java/com/facebook/login/WebLoginMethodHandler.java [hunk 1]
```diff
@@ -115,7 +115,9 @@ protected Bundle addExtraParameters(Bundle parameters, final LoginClient.Request
                 ServerProtocol.DIALOG_RETURN_SCOPES_TRUE);
         parameters.putString(
                 ServerProtocol.DIALOG_PARAM_AUTH_TYPE,
-                ServerProtocol.DIALOG_REREQUEST_AUTH_TYPE);
+                request.getAuthId()
```
Comment (abeatte) [inline facebook-common/src/main/java/com/facebook/login/WebLoginMethodHandler.java]: I think you mean getAuthType() here.

File: facebook-common/src/main/java/com/facebook/login/LoginManager.java [hunk 2]
```diff
@@ -368,6 +378,7 @@ public void logInWithReadPermissions(Activity activity, Collection<String> permi
         validateReadPermissions(permissions);
 
         LoginClient.Request loginRequest = createLoginRequest(permissions);
+        loginRequest.setAuthType(authType);
```
Comment (abeatte) [inline facebook-common/src/main/java/com/facebook/login/LoginManager.java]: I think it makes more sense to remove these and pass in 'authType' into the Request creation in line 493.

**Comment by sangpan** [comment]
> Look for this change to come out in the next SDK release. :-)

When can we change authType with public method on Android/iOS SDK?
2018 May? or June? or later?..

**Pull request closed without merging**